javascript/ql/lib/semmle/javascript/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -38,7 +38,10 @@ module CleartextLogging {
     MaskingReplacer() {
       this.isGlobal() and
       exists(this.getRawReplacement().getStringValue()) and
-      any(RegExpDot term).getLiteral() = this.getRegExp().asExpr()
+      exists(DataFlow::RegExpCreationNode regexpObj |
+        this.(StringReplaceCall).getRegExp() = regexpObj and
+        regexpObj.getRoot() = any(RegExpDot term)
```

Comment:
That doesn't look like the same behavior as previously.   
As I read it the previous behavior checks whether there is a `RegExpDot` anywhere in the regular expression (`getLiteral()` gets the literal for the root-term).  
  
But you've changed that to check whether the root itself is a `RegExpDot`.  

I think this change preserves the old behavior.  

```suggestion
        regexpObj.getRoot() = any(RegExpDot term).getRootTerm()
```

And a test where that makes a difference might look like: 
```JavaScript
console.log(password.replace(new RegExp("pre_._suf", "g"), "*")); // OK
```

Reply:
9ca0fe4cbf060fe8dc113257cd6d803d919e1a75